speechbrain/utils/optimizers.py

```diff
@@ -0,0 +1,23 @@
+"""Implements functions to avoid optimizing certain parameters
+
+Authors
+ * Titouan Parcollet 2023
+"""
+
+
+def rm_weight_decay_bias_and_norm_params(
```

Comment:
This function name suggests that it targets bias and norm parameters, and I suppose it most often ends up doing that, but the criterion is `len(param.shape) == 1`. Are you sure that is robust - will this sometimes catch other parameters? If I am thinking this out correctly, won't this essentially catch all vectors?

Is this something we always want to do for bias and norm params, I am not aware of the theory behind this?

Reply:
You are right, it will catch all parameters that are just vectors. But in most case, we may want to avoid pushing such parameters to 0. For BN and LN layers, it is detrimental to the results to do it. For biases, it simply is useless. ESPnet and K2 do this for instance. The len==1 is a global extension because I don't see any reason why someone would have scalar learnable parameters being pushed to 1. Note: This is only triggered if the yaml explicitly contains a self.weight_decay parameter .